src/t_cose_encrypt_enc.c

```diff
@@ -0,0 +1,579 @@
+        return(T_COSE_ERR_CBOR_FORMATTING);
+    }
+
+    cose_result = t_cose_crypto_encrypt(
+                       context->cose_algorithm_id,
+                       (struct q_useful_buf_c)
+                       {
+                          .ptr = context->key,
+                          .len = context->key_len
+                       },
+                       (struct q_useful_buf_c)
+                       {
+                          .ptr = context->nonce,
+                          .len = key_bitlen / 8
+                       },
+                       add_data_buf,
+                       payload,
+                       encrypted_payload,
+                       &ciphertext_length);
+
+    if (cose_result != T_COSE_SUCCESS) {
+        return(cose_result);
+    }
+
+    QCBOREncode_AddBytes(encrypt_ctx,
+                         (struct q_useful_buf_c)
+                            {
+                              .len = ciphertext_length,
+                              .ptr = encrypted_payload.ptr
+                            }
+                        );
+
+    return(T_COSE_SUCCESS);
+}
+
+enum t_cose_err_t
+t_cose_encrypt_add_recipient(struct t_cose_encrypt_enc_ctx* context,
+                             QCBOREncodeContext* encrypt_ctx,
+                             UsefulBufC* recipient_ctx)
+{
+    if (context->option_flags != T_COSE_OPT_COSE_ENCRYPT0) {
+        QCBOREncode_AddBuffer(encrypt_ctx,
+                              CBOR_MAJOR_NONE_TYPE_RAW,
+                              *recipient_ctx);
+        context->recipients++;
+        return(T_COSE_SUCCESS);
+    } else {
+        return(T_COSE_ERR_RECIPIENT_CANNOT_BE_ADDED);
+    }
+}
+
+
+enum t_cose_err_t
+t_cose_encrypt_hpke_create_recipient(struct t_cose_encrypt_recipient_hpke_ctx* context,
+                                     QCBOREncodeContext* EC)
+{
+    size_t                 key_bitlen;
+    size_t                 x_len;
+    int64_t                algorithm_id;
+    QCBORError             ret = QCBOR_SUCCESS;
+    UsefulBufC             scratch;
+    QCBOREncodeContext     ephemeral_key;
+    uint8_t                ephemeral_buf[100] = {0};
+    struct q_useful_buf    e_buf = {ephemeral_buf, sizeof(ephemeral_buf)};
+    uint8_t                encrypted_cek[PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(T_COSE_ENCRYPTION_MAX_KEY_LENGTH)];
+    size_t                 encrypted_cek_len = PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(T_COSE_ENCRYPTION_MAX_KEY_LENGTH);
+    UsefulBufC             cek_encrypted_cbor;
+    size_t                 pkR_len = PSA_EXPORT_PUBLIC_KEY_MAX_SIZE;
+    uint8_t                pkR[PSA_EXPORT_PUBLIC_KEY_MAX_SIZE] = {0};
+    size_t                 pkE_len = PSA_EXPORT_PUBLIC_KEY_MAX_SIZE;
+    uint8_t                pkE[PSA_EXPORT_PUBLIC_KEY_MAX_SIZE] = {0};
+    enum t_cose_err_t      return_value;
+    struct t_cose_crypto_hpke_suite_t hpke_suite;
+
+    if (context == NULL || EC == NULL) {
+        return(T_COSE_ERR_INVALID_ARGUMENT);
+    }
```

Comment:
I consider this check unnecessary and actually a waste of code.  

If every function in t_cose and CBOR checked for EC == NULL it would bloat the whole implementation a lot.